web/js/containers/sidebar/smart-handoff.js

```diff
@@ -0,0 +1,470 @@
+   * count vs selected count (that is if the bounding box has been enabled by the user)
+   * @param {*} currentExtent
+   */
+  async updateGranuleCount(currentExtent) {
+    const { selectedDate } = this.props;
+    const { selectedLayer, showBoundingBox } = this.state;
+
+    // Places the compoent state in a loading state; triggers {...} animation.
+    this.setState({ isSearchingForGranules: true });
+
+    const startDate = `${moment.utc(selectedDate).format('YYYY-MM-DD')}T00:00:00.000Z`;
+    const endDate = `${moment.utc(selectedDate).format('YYYY-MM-DD')}T23:59:59.999Z`;
+    const dateRange = `${startDate},${endDate}`;
+
+    let totalGranules = 0;
+    let selectedGranules = 0;
+
+    let urlTotalGranules = 'https://cmr.earthdata.nasa.gov/search/granules.json?'
+                + `temporal=${dateRange}&`
+                + `collection_concept_id=${selectedLayer.conceptId}&`
+                + 'include_facets=v2&'
+                + 'page_size=0';
+
+    if (selectedLayer.daynight) urlTotalGranules += `&day_night_flag=${selectedLayer.daynight}`;
+
+    let urlSelectedGranules = urlTotalGranules;
+
+    // Gets the total amount of granules that the layer has
+    totalGranules = await fetch(urlTotalGranules, { timeout: 5000 })
+      .then(async(response) => {
+        const result = await response.json();
+        return result.feed.facets.children[0].children[0].children[0].count;
+      })
+      .catch((error) => 0);
+
+    // Gets the total subset of granules that are within the defining bounding box
+    if (showBoundingBox) {
+      urlSelectedGranules += `&bounding_box=${currentExtent.southWest},${currentExtent.northEast}`;
+      selectedGranules = await fetch(urlSelectedGranules, { timeout: 5000 })
+        .then(async(response) => {
+          const result = await response.json();
+          return result.feed.facets.children[0].children[0].children[0].count;
+        })
+        .catch((error) => 0);
+    }
+
+    this.setState({ selectedGranules, totalGranules });
+    this.setState({ isSearchingForGranules: false });
+  }
+
+  /**
+   * Default render which displays the data-download panel
+   */
+  render() {
+    const {
+      screenWidth,
+      screenHeight,
+      activeLayers,
+      isActive,
+      showWarningModal,
+      selectedDate,
+    } = this.props;
+
+    const { selectedLayer, isSearchingForGranules } = this.state;
+    const {
+      selectedGranules, totalGranules, coordinates, currentExtent, showBoundingBox,
+    } = this.state;
+
+    // Determine if data-download 'smart-handoff' tab is activated by user
+    if (!isActive) return null;
+
+    const { boundaries } = this.state;
+    const {
+      x, y, x2, y2,
+    } = boundaries;
+
+    // Used to determine if the added smart-handoff modal should be shown
+    const { HIDE_EDS_WARNING } = safeLocalStorage.keys;
+    let showModal = safeLocalStorage.getItem(HIDE_EDS_WARNING);
+    showModal = true;
```

Comment:
You are always setting `showModal = true` here which means if the user checks the box "do not show this modal again" it has no effect and the modal always shows.

Reply:
Good catch. Forgot to remove this. Had this in there when I was editing some of the CSS for the modal and wanted it visible.